### DISCUSSION TITLE
Supabase realtime not working

### DISCUSSION THREAD

**Discussion by baloodevil** [supabase/supabase#21909] [Questions] [answered]
Wondering what I am missing here.  Does the channel `table-filter-changes` need to be set up somewhere?  Updating the status column in the table for this specific row does nothing.  ie: breakpoint inside on function doesn't get hit and status on the page doesn't change.

```Typescript
import { useEffect, useState } from "react";
import { createBrowserClient } from "@supabase/ssr";
import { DocumentRow } from "@/types/db";
import { DocumentStatus } from "@/types/enums";

export default function TestSomethingClient() {
  const [document, setDocument] = useState<Partial<DocumentRow>>({
    id: "d26d2f0e-5777-46d1-924c-1ed0e0d29938",
    status: DocumentStatus.None,
  });

  const supabase = createBrowserClient(process.env.NEXT_PUBLIC_SUPABASE_URL!, process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!);

  useEffect(() => {
    const updateListener = async () => {
      supabase
        .channel("table-filter-changes")
        .on(
          "postgres_changes",
          {
            event: "UPDATE",
            schema: "public",
            table: "documents",
            filter: `id=eq.${document.id}`,
          },
          (payload) => {
            const { status } = payload.new;
            const newDocument = { ...document, status };
            setDocument(newDocument);
          }
        )
        .subscribe();
    };
    updateListener();
  }, []);

  return <div>{document.status}</div>;
}
```

Realtime is enabled for this table...
![image](https://github.com/supabase/supabase/assets/11354890/d6e747b2-fa6c-46d1-9bae-24efb4a9b38a)

**Comment by GaryAustin1** [comment, marked as answer]
If you have select RLS on the table you have to meet it.
Your call at a quick glance looks fine.  The channel part is just an identifier if you want to close it later.
You could check realtime logs and also add .subscribe(status=>console.log(status))  to see if the connection is made.

**Reply by baloodevil** [reply]
It was RLS that was the problem.   For some reason this is not working `((auth.uid() = "userId") OR ('b209xxxx-xxxx-xxxx-xxxx-xxxxxxxxf0ec'::uuid = "userId"))` where my id is b209...   Not sure why that isn't working, but to do it right is to use [RBAC](https://supabase.com/docs/guides/auth/custom-claims-and-role-based-access-control-rbac) which conveniently talks about combining RBAC with RLS.  Thanks for the help!
